### PR TITLE
reference implementation of #1016

### PR DESCRIPTION
**Pull request by mvk**
sshagent plugin:
- manage setup of sshagent in bash session:
  - force opinionated agent management based on ~/.ssh/ file
    - support for shared file systems on different host (file name contains hostname)
  - check env for any live/dead agents
    if alive - use it
    otherwise - spawn our own

Signed-off-by: Max Kovgan <max@opsguru.io>

### CODE REVIEW

File: plugins/available/sshagent.plugin.bash [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    kill -9 "${agent_pid}" && echo "DONE" || echo "FAILED"
+    rm -f "${SSH_AGENT_ENV}"
+}
+
+
+function sshagent() {
+    [[ -v SSH_AGENT_ENV ]] \
+    || export SSH_AGENT_ENV="${HOME}/.ssh/agent_env.${HOSTNAME}"
+
+	case "${1}" in
```
Comment (nwinkler) [inline plugins/available/sshagent.plugin.bash]: The indentation seems to be off for the next couple of lines - can you please indent using 2 spaces per level?

File: plugins/available/sshagent.plugin.bash [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    _is_proc_alive_at_pid "${agent_pid}" \
+    || return # process is not alive - no problem
+
+    echo -e -n "Killing ssh-agent (pid:${agent_pid}) ... "
+    kill -9 "${agent_pid}" && echo "DONE" || echo "FAILED"
+    rm -f "${SSH_AGENT_ENV}"
+}
+
+
+function sshagent() {
```
Comment (nwinkler) [inline plugins/available/sshagent.plugin.bash]: Can you please add the required meta tags at the top of the function? Please see https://github.com/Bash-it/bash-it/blob/master/plugins/available/base.plugin.bash#L21 for examples.

File: plugins/available/sshagent.plugin.bash (hunk 1 above)
Comment (mvk) [inline plugins/available/sshagent.plugin.bash]: Will do, of course!

File: plugins/available/sshagent.plugin.bash [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  _is_proc_alive_at_pid "${agent_pid}" \
+  || return # process is not alive - no problem
+
+  echo -e -n "Killing ssh-agent (pid:${agent_pid}) ... "
+  kill -9 "${agent_pid}" && echo "DONE" || echo "FAILED"
+  rm -f "${SSH_AGENT_ENV}"
+}
+
+
+function sshagent() {
```
Comment (nwinkler) [inline plugins/available/sshagent.plugin.bash]: Can you please add the required meta tags at the top of the function? Please see https://github.com/Bash-it/bash-it/blob/master/plugins/available/base.plugin.bash#L21 for examples.

Comment (mvk) [inline plugins/available/sshagent.plugin.bash]: done.

**Comment by nwinkler** [comment]
Thanks!

**Pull request merged**